shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1842,28 +1946,36 @@ - (void)onOrientationPreferencesUpdated:(NSNotification*)notification {
   });
 }
 
+- (void)requestGeometryUpdateForWindowScene:(UIWindowScene*)windowScene API_AVAILABLE(ios(16.0)) {
+  UIWindowSceneGeometryPreferencesIOS* preference = [[[UIWindowSceneGeometryPreferencesIOS alloc]
+      initWithInterfaceOrientations:_orientationPreferences] autorelease];
+  [windowScene
+      requestGeometryUpdateWithPreferences:preference
+                              errorHandler:^(NSError* error) {
+                                os_log_error(OS_LOG_DEFAULT,
+                                             "Failed to change device orientation: %@", error);
+                              }];
+  [self setNeedsUpdateOfSupportedInterfaceOrientations];
+}
+
 - (void)performOrientationUpdate:(UIInterfaceOrientationMask)new_preferences {
   if (new_preferences != _orientationPreferences) {
     _orientationPreferences = new_preferences;
 
     if (@available(iOS 16.0, *)) {
+#if APPLICATION_EXTENSION_API_ONLY
+      if (self.windowSceneIfViewLoaded) {
+        [self requestGeometryUpdateForWindowScene:self.windowSceneIfViewLoaded];
+      }
+#else
       for (UIScene* scene in UIApplication.sharedApplication.connectedScenes) {
         if (![scene isKindOfClass:[UIWindowScene class]]) {
           continue;
         }
         UIWindowScene* windowScene = (UIWindowScene*)scene;
-        UIWindowSceneGeometryPreferencesIOS* preference =
-            [[[UIWindowSceneGeometryPreferencesIOS alloc]
-                initWithInterfaceOrientations:_orientationPreferences] autorelease];
-        [windowScene
-            requestGeometryUpdateWithPreferences:preference
-                                    errorHandler:^(NSError* error) {
-                                      os_log_error(OS_LOG_DEFAULT,
-                                                   "Failed to change device orientation: %@",
-                                                   error);
-                                    }];
-        [self setNeedsUpdateOfSupportedInterfaceOrientations];
+        [self requestGeometryUpdateForWindowScene:windowScene];
       }
+#endif
```

Comment:
Could we unify this code to have just populating a list of scenes into an ifdef, and then have the actual call to update the geometry be outside the ifdef, so we minimize code path branching.